registration-processor/pre-processor/registration-processor-introducer-validator-stage/src/main/java/io/mosip/registration/processor/stages/introducervalidator/IntroducerValidator.java

```diff
@@ -1,267 +1,308 @@
+import io.mosip.kernel.core.bioapi.exception.BiometricException;
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.util.StringUtils;
+import io.mosip.kernel.core.util.exception.JsonProcessingException;
+import io.mosip.registration.processor.core.code.RegistrationExceptionTypeCode;
+import io.mosip.registration.processor.core.constant.JsonConstant;
+import io.mosip.registration.processor.core.constant.MappingJsonConstants;
+import io.mosip.registration.processor.core.constant.ProviderStageName;
+import io.mosip.registration.processor.core.constant.RegistrationType;
+import io.mosip.registration.processor.core.exception.ApisResourceAccessException;
+import io.mosip.registration.processor.core.exception.IntroducerOnHoldException;
+import io.mosip.registration.processor.core.exception.PacketDecryptionFailureException;
+import io.mosip.registration.processor.core.exception.PacketManagerException;
+import io.mosip.registration.processor.core.exception.RegistrationProcessorCheckedException;
+import io.mosip.registration.processor.core.exception.ValidationFailedException;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.packet.dto.FieldValue;
+import io.mosip.registration.processor.core.status.util.StatusUtil;
+import io.mosip.registration.processor.core.util.RegistrationExceptionMapperUtil;
+import io.mosip.registration.processor.packet.manager.idreposervice.IdRepoService;
+import io.mosip.registration.processor.packet.storage.utils.BioSdkUtil;
+import io.mosip.registration.processor.packet.storage.utils.PriorityBasedPacketManagerService;
+import io.mosip.registration.processor.packet.storage.utils.Utilities;
+import io.mosip.registration.processor.status.code.RegistrationStatusCode;
+import io.mosip.registration.processor.status.dto.InternalRegistrationStatusDto;
+import io.mosip.registration.processor.status.dto.RegistrationStatusDto;
+import io.mosip.registration.processor.status.service.RegistrationStatusService;
+
+@Service
+public class IntroducerValidator {
+
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(IntroducerValidator.class);
+
+	public static final String INDIVIDUAL_TYPE_UIN = "UIN";
+
+	@Autowired
+	RegistrationExceptionMapperUtil registrationExceptionMapperUtil;
+
+	@Autowired
+	private PriorityBasedPacketManagerService packetManagerService;
+
+	@Autowired
+	private IdRepoService idRepoService;
+
+	@Autowired
+	RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	@Autowired
+	ObjectMapper mapper;
+
+	
+
+	@Autowired
+	private Utilities utility;
+	
+	@Autowired
+	private BioSdkUtil bioUtil;
+
+	@Value("#{T(java.util.Arrays).asList('${mosip.regproc.common.before-cbeff-others-attibute.reg-client-versions:}')}")
+	private List<String> regClientVersionsBeforeCbeffOthersAttritube;

```

Comment:
this change looks like the https://mosip.atlassian.net/browse/MOSIP-28478 at a high level, can you confirm the changes and jira id is matching ?

Reply:
https://mosip.atlassian.net/browse/MOSIP-28478  change is related to backward compatibility in abis handler stage and https://mosip.atlassian.net/browse/MOSIP-28557  change is related to backward compatibility in introducer validation stage